src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -248,29 +248,58 @@ public IObservable<Unit> UnmarkLocalBranch(ILocalRepositoryModel repository)
             });
         }
 
-        public IObservable<string> ExtractFile(ILocalRepositoryModel repository, string commitSha, string fileName)
+        public IObservable<string> ExtractFile(
+            ILocalRepositoryModel repository,
+            IModelService modelService,
+            string commitSha,
+            string fileName,
+            string fileSha)
         {
             return Observable.Defer(async () =>
             {
                 var repo = gitService.GetRepository(repository.LocalPath);
                 await gitClient.Fetch(repo, "origin");
-                var result = await gitClient.ExtractFile(repo, commitSha, fileName);
+                var result = await gitClient.ExtractFile(repo, commitSha, fileName) ??
+                             await modelService.GetFileContents(repository, commitSha, fileName, fileSha);
```

Comment:
Given that these are IO calls, there's a ton of different ways this can fail. We need to handle exceptions from these calls somewhere and fail sanely.